proposed/2022/dotnet-nuget-config-spec.md

```diff
@@ -0,0 +1,79 @@
+# NuGet configuration CLI for dotnet.exe
+
+- Author: [Heng Liu](https://github.com/heng-liu)
+- GitHub Issue [8420](https://github.com/NuGet/Home/issues/8420)
+
+## Problem Background
+
+Currently, there is no NuGet configuration CLI for dotnet.exe. It's inconvenient for NuGet users to know about the NuGet configuration file locations and figure out where is the merged configuration coming from. 
+
+## Who are the customers
+
+This feature is for dotnet.exe users.
+
+## Goals
+Design and implement `dotnet nuget config` command.
+
+## Non-Goals
+Design and implement `dotnet nuget config` command with commands other than `list`, E.g. add/update/delete
+
+## Solution
+The following command will be implemented in the `dotnet.exe` CLI.
```

Comment:
I really wish GitHub had threaded/resolvable conversations for comments without a specific file/line number 😔

I'd like to know what the design is for error scenarios, specifically when one or more of the XML files contains syntax errors (from an implementation point of view, this means that `Settings.Load*` apis will fail, so we can't use `loadedConfig.GetConfigFilePaths()`).

Is `dotnet nuget config list` going to report an error and fail to list all the files?  Will it at least tell us the filename of the XML that couldn't be parsed?  Or will all the XML files be listed despite the parsing failure?  Will it tell us which file(s) contain parsing errors?

Reply:
That's a good question!
I tested with NuGet.exe and dotnet.exe, and both of them are able to tell us tell us the filename of the XML that couldn't be parsed, as below:
```
nuget list source
NuGet.Config is not valid XML. Path: 'C:\Users\henli\Source\Repos\NuGet.Config'.
  The 'disabledPackageSources' start tag on line 19 position 4 does not match the end tag of 'configuration'. Line 20, position 3.
```
```
dotnet nuget list source
error: NuGet.Config is not valid XML. Path: 'C:\Users\henli\Source\Repos\NuGet.Config'.
error:   The 'disabledPackageSources' start tag on line 19 position 4 does not match the end tag of 'configuration'. Line 20, position 3.
```
So I think the dotnet nuget config list is able to do this as well.

If we'd like to list all config files despite the parsing failure, we'd consider if it worth doing.
Since running all nuget commands will encounter this invalid XML error, users need to fix this invalid XML before running any NuGet commands. So this seems to be a prerequisite and there is no harm if user needs to do this first before knowing the list of config files.
In addition, this error message clearly points out the file path and location,  so it should be easy for users to correct the invalid XML.

The only scenario needs to be considered is, with the invalid XML error, user should able to specifying a specific correct config file to run nuget commands. And if we list all config file locations despite the parsing failure, it might help people to find the path easier. But it seems "listing all config file locations despite the parsing failure" adds very limited value.

Those are my preliminary thoughts. Please let me know if you have any other thoughts :)

The other error scenario I'd like to talk about is, the specified working directory doesn't exist. Shall we go ahead listing user-wide and machine-wide config files? Or show a warning saying the working directory doesn't exist? Or both?